### PR TITLE
refactor(twap): cache fallback handler verification result

### LINKED ISSUE #2899: Cache the result of checking fallback handler
In `FallbackHandlerVerificationUpdater` once we detected, that fallback is set, we should:
 - cache the result in localStorage
 - don't execute `FallbackHandlerVerificationUpdater` logic again and use the cached value

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2899

In `FallbackHandlerVerificationUpdater` once we detected, that fallback is set, we should:
 - cache the result in `localStorage`
 - don't execute `FallbackHandlerVerificationUpdater` logic again and use the cached value

I also added `atomWithPartialUpdate` helper to reduce boilerplate code

  # To Test

1. Set fallback handler in TWAP

### CODE REVIEW

File: src/modules/limitOrders/state/limitRateAtom.ts [hunk 1]
```diff
@@ -29,12 +31,6 @@ export const initLimitRateState = () => ({
...
-export const limitRateAtom = atom<LimitRateState>(initLimitRateState())
-
-export const updateLimitRateAtom = atom(null, (get, set, nextState: Partial<LimitRateState>) => {
-  set(limitRateAtom, () => {
-    const prevState = get(limitRateAtom)
-
-    return { ...prevState, ...nextState }
-  })
-})
+export const { atom: limitRateAtom, updateAtom: updateLimitRateAtom } = atomWithPartialUpdate(
```
Comment (shoom3301) [inline src/modules/limitOrders/state/limitRateAtom.ts:34]: Just an example of `atomWithPartialUpdate` usage

File: src/utils/jotai/atomWithPartialUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { PrimitiveAtom, WritableAtom } from 'jotai'
+import { atom } from 'jotai/index'
+
+export function atomWithPartialUpdate<T>(anyAtom: PrimitiveAtom<T>): {
```
Comment (shoom3301) [inline src/utils/jotai/atomWithPartialUpdate.ts:4]: @alfetopito @nenadV91 @tukantje @anxolin pls, take this into account

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , I have set the handler
image
but the Advanced form proposes me to set it once again
image

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by alfetopito — APPROVED** [review]

File: src/utils/jotai/atomWithPartialUpdate.ts (hunk 2 above)
Comment (alfetopito) [reply src/utils/jotai/atomWithPartialUpdate.ts:4]: This is cool!

**Comment by elena-zh** [comment]
Hey @shoom3301 , the issue that I face is that I see 'Loading' button on the safe account where I 
1. set a handler once,
2.  then did a reset TX, 
3. then cleared out my local storage and refreshed the page

Also, the same result when I open the Safe app in another browser. 
image

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , I'm approving the current PR, however, I have created #2949 issue due to it is impossible to set the handler for the 2nd time. 
I'm afraid that the issue might be relevant for cases when the 1st transaction is cancelled/replaced/failed in the Safe.

File: src/utils/jotai/atomWithPartialUpdate.ts (hunk 2 above)
Comment (anxolin) [reply src/utils/jotai/atomWithPartialUpdate.ts:4]: great!

**Pull request merged**